src/essential_validator.py

```diff
@@ -309,10 +309,20 @@ def validate_data(self, file_info):
         columns = self.df.columns.tolist()
         empty_cols = [col for col in columns if not col or "Unnamed:" in col ]
         if empty_cols and len(empty_cols) > 0:
-            msg = f'“{file_info[FILE_NAME]}": some rows have extra columns.'
-            self.log.error(msg)
-            file_info[ERRORS].append(msg)
-            self.batch[ERRORS].append(msg)
+            for col in empty_cols:
+                extra_cell_list = self.df[self.df[col].notna()].index.astype(int).tolist()
+                if extra_cell_list and len(extra_cell_list):
+                    for index in extra_cell_list: 
+                        msg = f'“{file_info[FILE_NAME]}: line {index + 2}": some rows have extra columns.'
```

Comment:
Let's change the message to "extra columns/cells found", because we know the line number now.

Reply:
updated